core/cmd/lambda/main.go

```diff
@@ -95,11 +104,40 @@ func handler(
 			), err
 		}
 
+		userID := readUserID(req.Headers)
+		requestID := readRequestID(ctx)
+		if err := clientStorage.WritePrompt(
+			ctx, core.UserInput{
+				CallID: core.CallID{
+					RequestID: requestID,
+					UserID:    userID,
+				},
+				Prompt:    prompt,
+				Timestamp: time.Now().UTC(),
+			},
+		); err != nil {
+			log.Print("WritePrompt() error " + err.Error())
+		}
+
 		graph, err := clientModel.Do(ctx, prompt)
 		if err != nil {
 			return corsHeaders.setHeaders(parseClientError(err)), err
 		}
 
+		prediction, _ := json.Marshal(graph)
+		if err := clientStorage.WriteModelPrediction(
+			ctx, core.ModelOutput{
```

Comment:
suggestion for better readability, I would like to suggest creating a var to hold this core.ModelOutput value there.

Reply:
Thanks for your suggestion! I gladly apply it.